### PR TITLE
MAT-7684 UnsignedInt and MAT-7685 PositiveInt

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE PR

Jira Ticket: [MAT-7684](https://jira.cms.gov/browse/MAT-7684)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included
* [x] No extemporaneous files are included (i.e Complied files or testing results)
* [x] This PR is into the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/components/editTestCase/qiCore/LeftPanel/ElementsTab/builder/element/TypeEditor.tsx [hunk 1]
```diff
@@ -101,6 +102,30 @@ const TypeEditor = ({
...
+      case "unsignedInt":
+        return (
+          <IntegerComponent
+            canEdit={true}
+            structureDefinition={structureDefinition}
+            fieldRequired={required}
+            label={`Integer Field`}
+            onChange={onChange}
+            value={value}
+            signed={true}
```
Comment (nmorasb) [inline src/components/editTestCase/qiCore/LeftPanel/ElementsTab/builder/element/TypeEditor.tsx]: if this is for "unsignedInt" then shouldn't signed be false?
Looks like the main difference between unsignedInt and positiveInt is that unsignedInt includes 0, and positiveInt does not

Comment (sb-cecilialiu) [reply src/components/editTestCase/qiCore/LeftPanel/ElementsTab/builder/element/TypeEditor.tsx]: @nmorasb yeah you are right I'll make the change.

Comment (sb-cecilialiu) [reply src/components/editTestCase/qiCore/LeftPanel/ElementsTab/builder/element/TypeEditor.tsx]: @nmorasb I just pushed in the changed, thanks Nate!

**Review by jkotanchik-SB — DISMISSED** [review]

**Comment by nmorasb** [comment]
how will these refactors handle a signed integer? (Just a regular int that can be positive, 0 or negative) per https://jira.cms.gov/browse/MAT-7671

File: src/components/editTestCase/qiCore/LeftPanel/ElementsTab/builder/element/types/TypeComponentProps.ts [hunk 2]
```diff
@@ -5,5 +5,5 @@ export interface TypeComponentProps {
   onChange?: (nextValue: any) => void;
   structureDefinition: any;
   fieldRequired: boolean;
-  signed?: boolean;
+  unsignedInt?: boolean;
```
Comment (nmorasb) [inline src/components/editTestCase/qiCore/LeftPanel/ElementsTab/builder/element/types/TypeComponentProps.ts]: I don't think this is the right place for this prop, as TypeComponentProps is meant to be common to all of the type components and `unsignedInt` isn't a common prop. You can extend the TypeComponentProps interface for more specific component props
image

File: src/components/editTestCase/qiCore/LeftPanel/ElementsTab/builder/element/types/IntegerComponent.tsx [hunk 3]
```diff
@@ -11,14 +11,35 @@ const IntegerComponent = ({
   onChange,
   label = "Integer",
   structureDefinition,
-  signed = true,
+  unsignedInt = true,
```
Comment (nmorasb) [inline src/components/editTestCase/qiCore/LeftPanel/ElementsTab/builder/element/types/IntegerComponent.tsx]: how does this and all the other refactors related to unsigned vs positive int handle a regular/signed integer (as defined in [https://jira.cms.gov/browse/MAT-7671](https://jira.cms.gov/browse/MAT-7671))

File: src/components/editTestCase/qiCore/LeftPanel/ElementsTab/builder/element/types/TypeComponentProps.ts (hunk 2 above)
Comment (sb-cecilialiu) [reply src/components/editTestCase/qiCore/LeftPanel/ElementsTab/builder/element/types/TypeComponentProps.ts]: @nmorasb Ok thanks Nate, will do.

Comment (sb-cecilialiu) [reply src/components/editTestCase/qiCore/LeftPanel/ElementsTab/builder/element/types/TypeComponentProps.ts]: @nmorasb Just pushed in the changes

File: src/components/editTestCase/qiCore/LeftPanel/ElementsTab/builder/element/types/IntegerComponent.tsx (hunk 3 above)
Comment (sb-cecilialiu) [reply src/components/editTestCase/qiCore/LeftPanel/ElementsTab/builder/element/types/IntegerComponent.tsx]: @nmorasb Just pushed in the new changes thanks!

Comment (nmorasb) [reply src/components/editTestCase/qiCore/LeftPanel/ElementsTab/builder/element/types/IntegerComponent.tsx]: I'm still confused on how these changes will allow a user to enter a negative number? It seems like the previous validations that allowed someone to enter a regular integer (positive, 0 or negative) have been replaced with a minimum of either 0 or 1 depending on this `unsignedInt` prop. 
There are three cases this component needs to handle, right?

1. integer | A signed integer in the range −2,147,483,648..2,147,483,647
2. positiveInt | Any positive integer in the range 1..2,147,483,647
3. unsignedInt | Any non-negative integer in the range 0..2,147,483,647

Comment (nmorasb) [reply src/components/editTestCase/qiCore/LeftPanel/ElementsTab/builder/element/types/IntegerComponent.tsx]: Integer was done previously in 7671. 7684 was to add support for UnsignedInt, and 7685 was to add support for PositiveInt, but we also still need to keep support for a Signed Integer −2,147,483,648..2,147,483,647 even while adding support for PositiveInt and UnsignedInt, right?

Comment (sb-cecilialiu) [reply src/components/editTestCase/qiCore/LeftPanel/ElementsTab/builder/element/types/IntegerComponent.tsx]: @nmorasb Ok I see, I'll add signed type too thanks!

Comment (sb-cecilialiu) [reply src/components/editTestCase/qiCore/LeftPanel/ElementsTab/builder/element/types/IntegerComponent.tsx]: @nmorasb I just added Signed Integer support

**Review by nmorasb — APPROVED** [review]

**Pull request merged**